python/src/consume_message.py

```diff
@@ -38,9 +44,29 @@ def _read_sqs_message():
     try:
         message = message[0]
         print(datetime.datetime.utcnow(), message.body)
+
+
+        trace_header = message.attributes and message.attributes.get("AWSTraceHeader", None)
+
+        if trace_header:
+            global_sdk_config.set_sdk_enabled(True)
+
+            header = TraceHeader.from_header_str(trace_header)
+            trace_id = header.root
+            sampled = header.sampled
+
+            xray_recorder.begin_segment(
+                f"worker-{config.CLUSTER_ENV}-{config.SANDBOX_ID}",
+                traceid=trace_id,
+                sampling=sampled,
+                parent_id=header.parent,
+            )
+    
         body = json.loads(message.body)
         print(datetime.datetime.utcnow(), "Consumed a message from SQS.")
     except Exception as e:
+        xray_recorder.current_segment().add_exception(e, traceback.format_exc())
```

Comment:
If the xray_recorder.begin excepted will there be a current_segment ?

Reply:
Trying to understand how xray works

Reply:
No, but then that one will throw an exception and that will be caught by the worker.